storage/src/test/java/org/apache/kafka/storage/internals/log/OffsetMapTest.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.storage.internals.log;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import java.nio.ByteBuffer;
+import java.security.DigestException;
+import java.security.NoSuchAlgorithmException;
+import java.util.stream.IntStream;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class OffsetMapTest {
+
+    private static final int MEMORY_SIZE = 4096;
+
+    @ParameterizedTest
+    @ValueSource(ints = {10, 100, 1000, 5000})
+    public void testBasicValidation(int items) throws NoSuchAlgorithmException {
+        SkimpyOffsetMap map = new SkimpyOffsetMap(items * 48);
+        IntStream.range(0, items).forEach(i -> {
+            try {
+                map.put(key(i), i);
+            } catch (DigestException e) {
```

Comment:
We can replace lambda by loop to propagate checked exception. Or using `assertDoesNotThrow`